src/data/components/group.js

```diff
@@ -104,7 +111,24 @@ const Group = React.createClass({
 
   handleGroupCloseClick: function(event) {
     event.stopPropagation();
-    this.props.onGroupCloseClick(this.props.group.id);
+    this.setState({editing: false});
+    this.setState({closing: true});
+    this.setState({closeTimer: this.props.closeTimeout});
+
+    let group = this;
+
+    if (this.props.closeTimeout == 0) {
+      group.props.onGroupCloseClick(group.props.group.id);
+      return;
+    }
+
+    let timer = setInterval(function() {
+      group.setState({closeTimer: --group.state.closeTimer});
+      if (group.state.closeTimer <= 0) {
```

Comment:
Honestly, I don't think it's needed to show the actual timer here. I'd just use `setTimeout` with the specified delay and close the group after that. That countdown UI does not add anything useful in my opinion.